test/extensions/stats_sinks/hystrix/hystrix_test.cc

```diff
@@ -428,6 +438,42 @@ TEST_F(HystrixSinkTest, AddAndRemoveClusters) {
   // Check that old values of test_cluster2 were deleted.
   validateResults(cluster_message_map[cluster2_name_], 0, 0, 0, 0, 0, window_size_);
 }
+
+TEST_F(HystrixSinkTest, HystrixEventStreamHandler) {
+  InSequence s;
+  Buffer::OwnedImpl buffer = createClusterAndCallbacks();
+  // Register callback to sink.
+  sink_->registerConnection(&callbacks_);
+
+  // This value don't matter in handlerHystrixEventStream
+  absl::string_view path_and_query;
+
+  Http::HeaderMapImpl response_headers;
+  Stats::IsolatedStoreImpl listener_scope_;
+
+  NiceMock<AdminStreamMock> admin_stream_mock;
+
+  ON_CALL(admin_stream_mock, setEndStreamOnComplete(_)).WillByDefault(Return());
+  ON_CALL(admin_stream_mock, addOnDestroyCallback(_)).WillByDefault(Return());
+  ON_CALL(admin_stream_mock, getDecoderFilterCallbacks())
+    .WillByDefault(ReturnRef(callbacks_));
+
+  ASSERT_EQ(sink_->handlerHystrixEventStream(path_and_query, response_headers,
+                                             buffer, admin_stream_mock), Http::Code::OK);
+
+  // Check that response_headers has been set correctly
+  EXPECT_THAT(std::string(response_headers.ContentType()->value().getStringView()),
+              testing::HasSubstr("text/event-stream"));
```

Comment:
Why not just expect that they are equal?  Should there be other elements to these strings?

Reply:
I think I found some other test case where they were testing something similar and they used HasSubstr, but all of them except AccessControlAllowHeaders can be changed to equals (that one has a long-ish response. I wasn't sure if it was better to check equality against the entire response or just a substring of it)